packages/realm/src/IterableWeakSet.ts

```diff
@@ -0,0 +1,128 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { binding } from "./internal";
+
+/**
+ * Implements an iterable `WeakSet`.
+ * @internal
+ */
+export class IterableWeakSet<T extends object> implements Set<T> {
+  private internal: Set<binding.WeakRef<T>>;
```

Comment:
As you mentioned in the other PR, I might have used an array here instead. That could theoretically speed up deletion, but I'd need O(n) to find the object in any case, so I opted for something closer to the API it implements outwards.

Reply:
I also like that I can delete while iterating without having to keep track of the shifting offsets as I need to do with an array.

Reply:
But you still have O(n) here.